internal/backuprestore/backup.go

```diff
@@ -35,218 +34,45 @@ import (
-
-				switch currentBackupStatus {
-				case BackupPending:
-					pendingBackups = append(pendingBackups, existingBackup.Name)
-				case BackupFailed:
-					failedBackups = append(failedBackups, existingBackup.Name)
-				case BackupCompleted:
-					succeededBackups = append(succeededBackups, existingBackup.Name)
-				case BackupInProgress:
-					progressingBackups = append(progressingBackups, existingBackup.Name)
-				case BackupFailedValidation:
-					// Re-create the failed validation backup if it has been updated
-					if !equivalentBackups(existingBackup, backup) {
-						if err = h.Delete(ctx, existingBackup); err != nil {
-							return
-						}
-
-						var clusterID string
-						clusterID, err = getClusterID(ctx, h.Client)
-						if err != nil {
-							return
-						}
-						setBackupLabel(backup, map[string]string{clusterIDLabel: clusterID})
-						if err = h.Create(ctx, backup); err != nil {
-							return
-						}
-
-						progressingBackups = append(progressingBackups, existingBackup.Name)
-					} else {
-						failedValidationBackups = append(failedValidationBackups, existingBackup.Name)
-					}
-				}
-			}
-		}
-
-		if len(succeededBackups) == len(backupGroup) {
-			// The current backup group has done, work on the next group
-			continue
-
-		} else if len(failedBackups) != 0 {
-			status.Status = BackupFailed
-			status.Message = fmt.Sprintf(
-				"Failed backups: %s",
-				strings.Join(failedBackups, ","))
-			result.Requeue = false
-			return
-
-		} else if len(progressingBackups) != 0 {
-			status.Status = BackupInProgress
-			status.Message = fmt.Sprintf(
-				"Inprogress backups: %s",
-				strings.Join(progressingBackups, ","))
-			result.RequeueAfter = 2 * time.Second
-			return
-
-		} else if len(failedValidationBackups) != 0 {
-			status.Status = BackupFailedValidation
-			status.Message = fmt.Sprintf(
-				"Failed validation backups: %s. %s",
-				strings.Join(failedValidationBackups, ","),
-				"Please update the invalid backups.")
-			result.RequeueAfter = 1 * time.Minute
-			return
-
-		} else {
-			// Backup doesn't have any status, it's likely
-			// that the object storage backend is not available.
-			// Requeue to wait for the object storage backend
-			// to be ready.
-			status.Status = BackupPending
-			status.Message = fmt.Sprintf(
-				"Pending backups: %s. %s",
-				strings.Join(pendingBackups, ","),
-				"Wait for object storage backend to be available.")
-			result.RequeueAfter = 1 * time.Minute
-			return
-		}
-	}
-
-	status.Status = BackupCompleted
-	status.Message = "All Backups have completed"
-	result.Requeue = false
-	return
-}
-
-func equivalentBackups(backup1, backup2 *velerov1.Backup) bool {
+func EquivalentBackups(backup1, backup2 *velerov1.Backup) bool {
 	// Compare the specs only
 	return equality.Semantic.DeepEqual(backup1.Spec, backup2.Spec)
 }
 
-func (h *BRHandler) checkVeleroBackupProcessStatus(backup *velerov1.Backup) BackupPhase {
```

Comment:
Prefer you keep this function as-is. Have the log added in this function

Reply:
Updated